### PR TITLE
Brickwork structured ansatz

### PR DESCRIPTION
**Pull request by daito-quant**

### CODE REVIEW

File: packages/algo/quri_parts/algo/ansatz/__init__.py [hunk 1]
```diff
@@ -38,4 +39,5 @@
     "HardwareEfficientReal",
     "SymmetryPreserving",
     "SymmetryPreservingReal",
+    "Z2SymmetryPreserving",
```
Comment (ThomasenQunasys) [inline packages/algo/quri_parts/algo/ansatz/__init__.py]:
```suggestion
    "Z2SymmetryPreservingReal",
```

File: packages/algo/quri_parts/algo/ansatz/__init__.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
     TwoLocal,
     build_entangler_map,
 )
+from .z2_symmetry_preserving import Z2SymmetryPreserving
```
Comment (ThomasenQunasys) [inline packages/algo/quri_parts/algo/ansatz/__init__.py]:
```suggestion
from .z2_symmetry_preserving import Z2SymmetryPreservingReal
```

File: packages/algo/quri_parts/algo/ansatz/z2_symmetry_preserving.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    theta_0, theta_1 = circuit.add_parameters(
+        f"theta_{layer_index}_{j}", f"phi_{layer_index}_{j}"
+    )
+    circuit.add_gate(Rxx_gate((i, j), np.pi / 2))
+    circuit.add_ParametricRZ_gate(i, {theta_0: 1.0, theta_1: 1.0})
+    circuit.add_ParametricRZ_gate(j, {theta_0: -1.0, theta_1: 1.0})
+    circuit.add_gate(Rxx_gate((i, j), -np.pi / 2))
+
+
+class Z2SymmetryPreserving(ImmutableLinearMappedParametricQuantumCircuit):
```
Comment (ThomasenQunasys) [inline packages/algo/quri_parts/algo/ansatz/z2_symmetry_preserving.py]:
```suggestion
class Z2SymmetryPreservingReal(ImmutableLinearMappedParametricQuantumCircuit):
```

File: packages/algo/tests/algo/ansatz/test_z2_symmetry_preserving.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
+from typing import Callable, Sequence
+
+import numpy as np
+
+from quri_parts.algo.ansatz import Z2SymmetryPreserving
```
Comment (ThomasenQunasys) [inline packages/algo/tests/algo/ansatz/test_z2_symmetry_preserving.py]:
```suggestion
from quri_parts.algo.ansatz import Z2SymmetryPreservingReal
```

File: packages/algo/tests/algo/ansatz/test_z2_symmetry_preserving.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    ansatz: ImmutableLinearMappedParametricQuantumCircuit,
+    test_circuit_builder: Callable[[Sequence[float]], QuantumCircuit],
+) -> None:
+    params_list = list(2.0 * np.pi * np.random.random(ansatz.parameter_count))
+    circuit = ansatz.bind_parameters(params_list)
+    test_circuit = test_circuit_builder(params_list)
+    assert circuit.gates == test_circuit.gates
+
+
+def test_Z2SymmetryPreserving() -> None:
```
Comment (ThomasenQunasys) [inline packages/algo/tests/algo/ansatz/test_z2_symmetry_preserving.py]:
```suggestion
def test_Z2SymmetryPreservingReal() -> None:
```

File: packages/algo/quri_parts/algo/ansatz/z2_symmetry_preserving.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    """z2_symmetry_preserving.
```
Comment (ThomasenQunasys) [inline packages/algo/quri_parts/algo/ansatz/z2_symmetry_preserving.py]:
```suggestion
    """z2 symmetry preserving real ansatz.
```

**Review by ThomasenQunasys — COMMENTED** [review]
Hi @daito-quant , just one last change (you can incorporate it by accepting the suggested changes above).

In the future we may add a complex valued version of this ansatz, so for now I want to add the suffix `Real` to this version so we don't have to change it later. Sorry for the last minute request.

Also, please sign the CLA. You can just click the link provided ;)

**Comment by daito-quant** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by ThomasenQunasys — APPROVED** [review]

**Pull request merged**